src/node/context.h

```diff
@@ -34,6 +35,7 @@ struct NodeContext {
     std::unique_ptr<BanMan> banman;
     std::unique_ptr<interfaces::Chain> chain;
     std::vector<std::unique_ptr<interfaces::ChainClient>> chain_clients;
+    CScheduler* scheduler{nullptr};
```

Comment:
Why not unique_ptr?

Reply:
Maybe I get it? Who owns the scheduler originally?

Reply:
`init.cpp` instantiates the pointer as a static var. I'm trying to dig into better understand the memory management & if this can be a `unique_ptr`, but dont fully get it yet

Reply:
if it's a static var in init.cpp it can't be a unique_ptr, because unique_ptr owns it's memory.

However, you could potentially figure out how to pass in a reference to the node context constructor. Otherwise the pointer is fine...